src/box/lua/tuple.c

```diff
@@ -779,12 +777,17 @@ box_lua_tuple_init(struct lua_State *L)
 	luaL_findtable(L, LUA_GLOBALSINDEX, "box.internal", 1);
 	luaL_newmetatable(L, tuplelib_name);
 	luaL_setfuncs(L, lbox_tuple_meta, 0);
+	lua_pushcfunction(L, lbox_tuple_new);
+	lua_setfield(L, -2, "new");
 	lua_setfield(L, -2, "tuple");
 	lua_pop(L, 1); /* box.internal */
 	luaL_register_type(L, tuple_iteratorlib_name,
 			   lbox_tuple_iterator_meta);
+	/*
+	 * Lua part relies on the existence of this table, so we need to make
+	 * sure it exists or create it on the way.
+	 */
```

Comment:
Nit: Better set it where it's actually needed, i.e. in Lua:

```diff
diff --git a/src/box/lua/tuple.lua b/src/box/lua/tuple.lua
index 33aa444849b4..2903051b8cd9 100644
--- a/src/box/lua/tuple.lua
+++ b/src/box/lua/tuple.lua
@@ -303,6 +303,8 @@ local methods = {
     ["tomap"]       = internal.tuple.tuple_to_map;
 }
 
+box.tuple = {}
+
 -- Aliases for tuple:methods().
 for k, v in pairs(methods) do
     box.tuple[k] = v
```

Reply:
This won't work, since the tuple format module is initialized before the tuple module, and it needs to create the `box.tuple.format` table.

Reply:
Dropped this hunk.